src/main/java/org/sagebionetworks/web/client/widget/doi/CreateOrUpdateDoiModal.java

```diff
@@ -62,21 +64,26 @@ public void configureAndShow(String objectId, ObjectType objectType, Long versio
 		this.entityUpdatedHandler = entityUpdatedHandler;
 		doi = new Doi();
 		getExistingDoi(objectId, objectType, versionNumber);
-		view.show();
 	}
 
 	public void getExistingDoi(String objectId, ObjectType objectType, Long objectVersion) {
 		javascriptClient.getDoi(objectId, objectType, objectVersion).addCallback(new FutureCallback<Doi>() {
 			@Override
 			public void onSuccess(@NullableDecl Doi doi) {
 				populateForms(doi);
+				view.show();
 			}
 
 			@Override
 			public void onFailure(Throwable t) {
-				doi.setObjectId(objectId);
-				doi.setObjectType(objectType);
-				doi.setObjectVersion(objectVersion);
+				if (t instanceof NotFoundException) {
+					doi.setObjectId(objectId);
+					doi.setObjectType(objectType);
+					doi.setObjectVersion(objectVersion);
+					view.show();
+				} else {
+					popupUtilsView.showErrorMessage(DOI_SERVICES_UNAVAILABLE_AT_THIS_TIME);
```

Comment:
always include error message coming from the backend (we're losing information here).  please replace with 
popupUtilsView.showErrorMessage(t.getMessage());